api/handlers/carts.js

```diff
@@ -0,0 +1,320 @@
+                    statusCode: 200,
+                    body: JSON.stringify({newQuantity: newQuantity})
+                }
+            }
+
+        }
+    } catch(error){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 500,
+            body: JSON.stringify({error: 'Internal server error'})
+        }
+    }
+    }
+
+//get cart by userId
+module.exports.getCartByUserID = async (event) => {
+try{
+    const userId = event.pathParameters && event.pathParameters.user_id
+    if(!userId){
+        return{
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 400,
+            body: JSON.stringify({message: 'invalid user id'})
+        }
+    } 
+  else{
+    const response = await pool.query('SELECT * FROM carts WHERE user_id = $1', [userId]);
+    return {
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+           
+          },
+        statusCode: 200,
+        body: JSON.stringify(response.rows)
+    }
+}
+} catch(error){
+    return {
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+           
+          },
+        statusCode: 500,
+        body: JSON.stringify({error: 'Internal server error'})
+    }
+}
+}
+
+//delete cart item
+module.exports.deleteCartItem = async (event) => {
+    try {
+    const userId = event.queryStringParameters && event.queryStringParameters.user_id
+    const productId = event.queryStringParameters && event.queryStringParameters.product_id
+    if(!userId || !productId){
+        return{
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 400,
+            body: JSON.stringify({message: 'invalid user id or product ID'})
+        }
+    } else{
+        const result = await pool.query('DELETE FROM carts WHERE user_id = $1 AND product_id = $2 RETURNING *', [userId, productId])
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 200,
+            body: JSON.stringify(result.rows)
+        }
+    }
+} catch(error){
+    return {
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+           
+          },
+        statusCode: 500,
+        body: JSON.stringify({error: 'Internal server error'})
+    }
+}
+
+}
+
+//put route for cart, specifically re-setting quantity
+module.exports.put = async (event) => {
+    try {
+    const userId = event.queryStringParameters && event.queryStringParameters.user_id
+    const productId = event.queryStringParameters && event.queryStringParameters.product_id
+    const quantity = event.queryStringParameters && event.queryStringParameters.quantity
+    const parsedQuantity = parseInt(quantity, 10)
+    if(!userId || !productId || !quantity){
```

Comment:
Is it valid for the frontend code ever pass in a zero here, like if someone types in a 0 quantity because they decided they don't want that item? 

If so this error message will be displayed because `!0` is `true`. That's OK if you don't want to accept 0, but if you do want to accept 0 it is a problem.